app/src/main/java/space/taran/arknavigator/mvp/model/repo/preview/PreviewGenerators.kt

```diff
@@ -45,26 +47,64 @@ object PreviewGenerators {
             Log.d(PREVIEWS, "Thumbnail for image $path generated in $time1 ms")
             return
         }
+
         val ext = extension(path)
         generatorsByExt[ext]?.let { generator ->
             val time2 = measureTimeMillis {
-                val thumbnail = if (ext == "txt") {
-                    resizePreviewToThumbnail(generator(path))
-                } else {
-                    val preview = generator(path)
-                    storePreview(previewPath, preview)
-                    resizePreviewToThumbnail(preview)
-                }
+                val preview = generator(path)
+                storePreview(previewPath, preview)
+                val thumbnail = resizePreviewToThumbnail(preview)
```

Comment:
We shouldn't call this for text. Text should fall into the same branch as images (images don't have previews, the same as text resources). I think code above should be adapted to include text too:
```
if (ImageKindFactory.isValid(path)) { // || PlainTextKindfactory.isValid(path)
...
}
```

Reply:
If we put this then I have to check again for txt file so I can called `            TxtPreviewGenerator.generate(path, THUMBNAIL_SIZE)
` to create bitmap using some text from the file. instead this, I will put `PlainTextKindfactory.isValid(path)` in else if condition.